New package: Composer.Composer version 2.8.6

Dragon1573 opened the pull request:
## Checklist for Pull Requests

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Resolve #242786

## Manifests

- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [ ] Does your manifest conform to the [1.10 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.10.0)?
  - [x] Or does it conform to the [1.9 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.9.0)?

Note: `<path>` is the directory's name containing the manifest you're submitting.

## Manual validation

![image](https://github.com/user-attachments/assets/c92ef9c5-9152-474b-bf8e-3b45b2ca6a61)
![image](https://github.com/user-attachments/assets/263b77d8-0090-4b0d-ba93-49353c930143)

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/242896)

Dragon1573 committed:
New package: Composer.Composer version 2.8.6

Signed-off-by: Dragon1573 <49941141+Dragon1573@users.noreply.github.com>

mdanish-kh commented:
How is this different from #204173? This still looks like a scripted application as per your screenshot

[Policy] Blocking-Isuse
[Policy] Scripted-Application

Dragon1573 commented:
> How is this different from #204173? This still looks like a scripted application as per your screenshot

It's entirely different from that!

`Composer v2.8.6` now provides an **InnoSetup install wizard** for the `composer` and `composer.bat`. They were added to `$env:PATH` by the install wizard instead of WinGet. NO symbolic links was created by WinGet in default `${env:ProgramFiles}\WinGet\Links\` folder.

I think this is acceptable.

Dragon1573 committed:
Update manifest: Composer.Composer version 2.8.6

- Add "7zip.7zip" as a dependency for downloading and extracting dependencies.
- Update the architecture from "x86" to "x64" for all scopes.

Signed-off-by: Dragon1573 <49941141+Dragon1573@users.noreply.github.com>

Dragon1573 commented:
## ✅ Manual validation

![image](https://github.com/user-attachments/assets/a9e76097-54ac-4467-bc16-51e518b051b2)
![image](https://github.com/user-attachments/assets/25c4941b-faab-452b-adff-94cffafbb6eb)

-----

Can't sure why it failed with <https://github.com/microsoft/winget-pkgs/labels/Validation-Installation-Error> and <https://github.com/microsoft/winget-pkgs/labels/Validation-Unattended-Failed>

Dragon1573 commented:
> The package manager bot determined there was an issue with the pull request. Make sure the manifest files are under the manifests\partition\publisher\appname\version directory and only one package version is being modified in your PR. The partition of the path must be the first letter of the publisher in lower-case.
> 
> For details on the error, see the details link below in the build pipeline.
> 
> Template: msftbot/validationError/pullRequest

Can't figure out why. 😕 

```text
22:57:18 D:\...\winget-pkgs  [issues/242786 ≡] 0ms pwsh$ git last    
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
commit d225849af2b9e944d5729d042583e112184c2f6e (HEAD -> issues/242786, origin/issues/242786)
Author: Dragon1573 <49941141+Dragon1573@users.noreply.github.com>
Date:   Thu Mar 27 21:51:53 2025 +0800

    Update manifest: Composer.Composer version 2.8.6
    
    - Add "7zip.7zip" as a dependency for downloading and extracting dependencies.
    - Update the architecture from "x86" to "x64" for all scopes.
    
    Signed-off-by: Dragon1573 <49941141+Dragon1573@users.noreply.github.com>

22:57:22 D:\...\winget-pkgs  [issues/242786 ≡] 506ms pwsh$ winget validate .\manifests\c\Composer\Composer\2.8.6\
清单具有以下未验证的依赖项;确保它们有效：
  - 程序包
      PHP.PHP.8.4
      7zip.7zip
清单验证成功。
```

mdanish-kh commented:
> Composer v2.8.6 now provides an InnoSetup install wizard for the `composer` and `composer.bat`. They were added to `$env:PATH` by the install wizard instead of WinGet. NO symbolic links was created by WinGet in default `${env:ProgramFiles}\WinGet\Links\` folder.

> I think this is acceptable.


I believe the repository policies prohibit scripts both as the executable and as the installer. This is the case of the executable being a script file (`.bat`) even though the installer was a `.exe` inno setup.

https://github.com/microsoft/winget-pkgs/blob/50fd2370d3ffec3ef5566d05fd0888e777a597c4/.github/policies/labelAdded.scriptedApplication.yml#L24-L30

[comment]: <[Policy] Blocking-Issue>

Dragon1573 commented:
> I believe the repository policies prohibit scripts both as the executable and as the installer. This is the case of the executable being a script file (`.bat`) even though the installer was a `.exe` inno setup.

😕 Okay ...

> [!IMPORTANT]
> 
> Close with reason: **Repository policy violation**.
>
> Submitting packages those uses scripts as executables are prohibited, EVEN THOUGH they were installed via any kinds of supported install wizards.
